### ISSUE TITLE
SMS text notification issue

### ISSUE THREAD

**Issue by Phaedog** [GrapheneOS/os-issue-tracker#817]
SMS text notifications appear and sound at the time when phone is unlocked.  DND is off. Battery saver is off. Notifications on lock screen to "show". Tried different permissions and configuration options. I'm using a new google pixel 5a5g. Called the carrier. Reset the WiFi, Mobile and Bluetooth via the "reset option". It's not the carrier. I have a website where I can send myself a message and I get the massage notification at whatever interval I choose to unlock my phone.  I see there have been a few issues reported about this.

**Comment by Phaedog** [comment]
What's the status of this?

**Comment by lbdroid** [comment]
I don't think this is upstream. I did not experience it with aosp12r12 on raven.

**Comment by frogwoosh** [comment]
This is very much a problem for me, my phone can no longer consistently notify me :(

**Comment by ayo-x** [comment]
This is an issue since Android 10. Both SMS notifications and alarms are unreliable.

**Comment by thestinger** [comment]
Not the case for the vast majority of our users.

**Comment by tsaoi** [comment]
@thestinger yes this is a known ongoing issue and appears to be related to #510

**Comment by lbdroid** [comment]
From what I can tell, it appears to just be that the device goes to sleep before the notification is triggered. As soon as the device is woken up, the message notification fires off. If most people are using RCS or just a different messaging application, this could be the reason why they don't experience this issue -- different balancing of wake locks. Its been an issue for me under different devices, different AOSP versions, and different distributions (i.e., also experienced under Proton AOSP11 on Pixel2XL, whereas now its under Graphene AOSP12 on Pixel6Pro).

**Comment by whats-a-diorama** [comment]
I've been having this issue for some time now. Basically, when my phone was active, I would get Messaging notifications immediately. But when it was locked/idle, they often wouldn't appear at all until I woke my phone up and unlocked it. So SMS messages could go unnoticed for quite a while.

I had also checked all the usual suspects: Battery Saver was off, Battery Manager was off, Messaging was set to Unrestricted Battery Use, etc. I even made sure Messaging's Background Data and Unrestricted Data Usage were both turned on, just in case. But none of those ever made a difference.

It might also be worth noting that Messaging is the only app that I've had this issue with. For example, I always get immediate notifications from Signal upon receiving a message, OpenTasks notifications always appear right on time, etc.

But I recently tried something, and it appears to have totally resolved the issue for me.

I decided to look at the deviceidle whitelist for my device via adb, and I found that `com.android.messaging` was listed for both `system-excidle` and `system`, but not for `user`. All of my apps that I've installed myself of course exist in `user`, and my notifications for those have always worked flawlessly. So I tried adding Messaging to `user`, and sure enough, I'm consistently getting all of my notifications immediately now. Even if I haven't touched my phone for hours.

I'm not an Android expert, and I'm guessing that adding Messaging to the `user` whitelist is not the proper solution here. So I guess I'm basically just reporting that I too have been experiencing this issue and that it's specific only to Messaging's notifications, and also providing what little insight I can based on how I managed to resolve it for myself.

**Comment by tsaoi** [comment]
> I'm not an Android expert, and I'm guessing that adding Messaging to the `user` whitelist is not the proper solution here. So I guess I'm basically just reporting that I too have been experiencing this issue and that it's specific only to Messaging's notifications, and also providing what little insight I can based on how I managed to resolve it for myself.

Thanks for this insight! Hopefully @thestinger can incorporate some fixes in an upcoming release! :)

**Comment by thestinger** [comment]
I don't think that means anything other than the fact that it wasn't an exception created by a user.

I'm only one of the developers who works on GrapheneOS and there is no one working on this area at all.

**Comment by thestinger** [comment]
There will not be any changes to the Messaging app in the near future. Posting content free replies here will result in the thread being locked or deleted since it takes away from development time.

**Comment by lbdroid** [comment]
Nevertheless, after quickly testing this, it appears to work.

```dumpsys deviceidle whitelist +com.android.messaging```

**Comment by lbdroid** [comment]
@Phaedog its a command you type into a terminal. I.e., "adb shell" from your computer hooked up to the phone via USB wire.

**Comment by Phaedog** [comment]
> @Phaedog its a command you type into a terminal. I.e., "adb shell" from your computer hooked up to the phone via USB wire.

Thanks for finding a solution for this. The messaging works lightning fast now. Will I have to program this every time they push out a new update?

**Comment by muhomorr** [comment]
The next time this issue occurs, capture a logcat. Current GrapheneOS version enables logging in the `DeviceIdleController` which may help to find the root cause.
